classes/01class/exercises/c01-git04/luizfds/c01-git04-1.txt

```diff
@@ -0,0 +1,57 @@
+Q1. Let's suppose you remotely pushed your my_env.txt file. A colleague asks you to remove this information from Git. What's your colleague worried about?
+He is worried that anyone with access to the repository can have access to sensitive information added in the file.
+
+Q2. If you modify the file in your workspace, then push it, will it be enough? (It's not). Why?
+No, it will not be enough. As we saw in the previous exercise, Git creates a copy of the file and adds it to git history.
+
+Q3. If you delete the file and push it, then create a new one with the rest of the information, is it enough? (It's not). Why?
+No, because git will just add a new commit in the history with a new version of the file, the previous version will still be present in the history. 
+
+Q4. How to fix this? How do you remove something from Git history?
+To fix this and entirely remove the unwanted file from our repository's history we can use either the git filter-branch command or the BFG Repo-Cleaner open source tool.
+
+Q5. Which commands would you use? What are the consequences for other developers?
+The git filter-branch command and the BFG Repo-Cleaner rewrite git repository's history, which changes the SHAs for existing commits that you alter and any dependent commits. Changed commit SHAs may affect open pull requests in your repository. So it’s recommended merging or closing all open pull requests before removing files from your repository.
```

Comment:
Very good!

Instead of using more complex tools like `filter-branch` (that might rewrite the whole history), you can use `git rebase -i --root` and `drop` the commits you want (or just delete the commit line).
Then `force push it`.

It will remove the commit from the remote history, although you can still check it out in your local repo (requires deeper local clean up).